### PR TITLE
[Settings v2] Use history back for settings v2 pages navbar

### PR DESCRIPTION
**Pull request by IniZio**
ref DEV-2094

Initially wanted to hide back button when there is no prior history. However `window.history.length` actually always return all backword + forward navigations. 

We cannot reliably check if current page is the last item in the stack

### CODE REVIEW

**Comment by louischan-oursky** [comment]
Build failed.

File: resources/authgear/templates/en/web/authflowv2/__navbar.html [hunk 1]
```diff
@@ -2,13 +2,13 @@
 <nav class="navbar">
   <h1 class="navbar__title">{{ $.Title }}</h1>
 
-  <a
-    class="navbar__item inline-flex items-center justify-center gap-x-3"
+  <button
+    class="navbar__item"
```
Comment (louischan-oursky) [inline resources/authgear/templates/en/web/authflowv2/__navbar.html:6]: When javascript is disabled, this button has no effect. So when javascript is disabled, it should be hidden. The idiomatic way to do this is to add `hidden` to the class, so initially it is hidden. When the controller connects, you run `this.element.classList.remove("hidden")` to make the button visible.

**Review by louischan-oursky — CHANGES_REQUESTED** [review]

File: resources/authgear/templates/en/web/authflowv2/__navbar.html (hunk 1 above)
Comment (louischan-oursky) [reply resources/authgear/templates/en/web/authflowv2/__navbar.html:6]: Added this to https://github.com/authgear/authgear-server/issues/3626#:~:text=Always%20think%20about%20when%20the%20case%20when%20javascript%20is%20disabled

Comment (IniZio) [reply resources/authgear/templates/en/web/authflowv2/__navbar.html]: Fixed up

**Review by louischan-oursky — APPROVED** [review]

**Pull request merged**